src/main/integrations/child-process.ts

```diff
@@ -0,0 +1,113 @@
+import { addBreadcrumb, captureMessage, getCurrentHub } from '@sentry/core';
+import { NodeClient } from '@sentry/node';
+import { Integration, Severity } from '@sentry/types';
+
+import { ALL_REASONS, ExitReason, onChildProcessGone, onRendererProcessGone } from '../electron-normalize';
+import { ElectronMainOptions } from '../sdk';
+
+interface ChildProcessOptions<T> {
+  /** Child process events that generate breadcrumbs */
+  breadcrumbs: T;
+  /** Child process events that capture events */
+  capture: T;
+}
+
+const DEFAULT_OPTIONS: ChildProcessOptions<ExitReason[]> = {
+  breadcrumbs: ALL_REASONS,
+  capture: ['abnormal-exit', 'launch-failed', 'integrity-failure'],
+};
+
+/** Gets message and severity */
+function getMessageAndSeverity(reason: ExitReason, proc?: string): { message: string; level: Severity } {
+  const message = `'${proc}'' process exited with '${reason}'`;
+
+  let level = Severity.Debug;
+
+  switch (reason) {
+    case 'abnormal-exit':
+    case 'killed':
+      level = Severity.Warning;
+      break;
+    case 'crashed':
+    case 'oom':
+    case 'launch-failed':
+    case 'integrity-failure':
+      level = Severity.Critical;
+  }
+
+  return { message, level };
```

Comment:
this is a nit, so feel free to ignore. IMO the immutable returns are clearer to understand.

```suggestion
  switch (reason) {
    case 'abnormal-exit':
    case 'killed':
      return { message, level: Severity.Warning};
    case 'crashed':
    case 'oom':
    case 'launch-failed':
    case 'integrity-failure':
      return { message, level: Severity.Critical};
    
    return return { message, level: Severity.Debug};
  }

  return { message, level };
```

Reply:
No I agree here!